src/components/molecules/Jukebox/Jukebox.tsx

```diff
@@ -0,0 +1,169 @@
+  const [filteredUser] = recentVenueUsers.filter(({ id }) => id === userId);
+  const tableRef = filteredUser?.data?.[venue.name]?.table ?? "";
+
+  const { sendJukeboxMsg, messagesToDisplay } = useJukeboxChat({
+    venueId: venue.id,
+    tableId: tableRef,
+  });
+  const filteredMessages = messagesToDisplay.filter(
+    ({ tableId }) => tableId === tableRef
+  );
+
+  useEffect(() => {
+    const [lastMessage] = messagesToDisplay.slice(-1);
+    let urlToEmbed = lastMessage?.text;
+    if (
+      urlToEmbed?.includes(YOUTUBE_SHORT_URL_STRING) &&
+      isValidUrl({ url: urlToEmbed, isStrict: true })
+    ) {
+      urlToEmbed = getYoutubeEmbedFromUrl(lastMessage?.text);
+    }
+    if (isValidUrl({ url: urlToEmbed, isStrict: true })) {
+      updateIframeUrl(urlToEmbed);
+    }
+  }, [messagesToDisplay, updateIframeUrl]);
+
+  const sendMessageToChat = handleSubmit(async ({ jukeboxMessage }) => {
+    setMessageSending(true);
+
+    await sendJukeboxMsg({ message: jukeboxMessage });
+    reset();
+  });
+
+  // @debt replace with useDebounce
+  // This logic disallows users to spam into the chat. There should be a delay, between each message
+  useEffect(() => {
+    if (!isSendingMessage) return;
+
+    const timeoutId = setTimeout(() => {
+      setMessageSending(false);
+    }, CHAT_MESSAGE_TIMEOUT);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [isSendingMessage]);
+
+  const isBtnDisabled = !chatValue || isSendingMessage;
+  const submitButtonClasses = classNames("Jukebox__buttons-submit", {
+    "Jukebox__buttons-submit--active": !isBtnDisabled,
+  });
+
+  const messagesEndRef = useRef<null | HTMLDivElement>(null);
+
+  const scrollToBottom = () => {
+    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
+  };
+
+  useEffect(scrollToBottom, [messagesToDisplay]);
+
+  const { openUserProfileModal } = useProfileModalControls();
+
+  const jukeboxChatMessages = useMemo(
+    () =>
+      filteredMessages?.map((msg) => {
+        const textMessage = `${
+          isValidUrl({ url: msg.text, isStrict: true })
+            ? `changed video source to ${msg.text}`
+            : msg.text
+        }`;
+        return (
+          <div key={`${msg.id}`} className="Jukebox__chat-messages">
+            <span
+              className="Jukebox__chat-author"
+              onClick={() => openUserProfileModal(msg.author)}
+            >
+              {msg.author.partyName}{" "}
+            </span>
+            <span>{textMessage}</span>
+          </div>
+        );
+      }),
+    [filteredMessages, openUserProfileModal]
+  );
+
+  return (
+    <>
+      <div className="Jukebox__container">
+        <div className="Jukebox__chat">
+          <span className="Jukebox__chat-messages">
+            JUKEBOX RULES: There’s no queue system. If you post up your link,
+            it’ll play. Be courteous people, do not post a new link until the
+            other one is finished!
+          </span>
```

Comment:
let's give this text a bit opacity to distinguish it from users' messages. (opacity: 0.8 I guess should look good)

Reply:
fixed f0bbdc8